core-bundle/tests/EventListener/MergeHttpHeadersListenerTest.php

```diff
@@ -236,6 +238,37 @@ public function testDoesNotMergeCacheControlHeaders(): void
         $this->assertSame('no-cache, private', $response->headers->get('Cache-Control'));
     }
 
+    public function testServiceIsResetable(): void
+    {
+        $response = new Response();
+        $response->headers = $this->createMock(ResponseHeaderBag::class);
+
+        $response->headers
+            ->expects($this->exactly(2))
+            ->method('set')
+            ->with('foo', 'Bar')
+        ;
+
+        $framework = $this->createMock(ContaoFramework::class);
+        $framework
+            ->expects($this->exactly(3))
+            ->method('isInitialized')
+            ->willReturn(true)
+        ;
+
+        $headerStorage = new MemoryHeaderStorage(['Foo: Bar']);
+
+        $listener = new MergeHttpHeadersListener($framework, $headerStorage);
+
+        $this->assertInstanceOf(ResetInterface::class, $listener);
+
+        $listener->onKernelResponse($this->getResponseEvent($response));
+        $listener->onKernelResponse($this->getResponseEvent($response));
+
+        $listener->reset();
+        $listener->onKernelResponse($this->getResponseEvent($response));
```

Comment:
There is no assertion here. Based on the method name, I assume you are testing that no exception is thrown? If so, please add the following line:

```php
$this->addToAssertionCount(1);  // does not throw an exception
```

Reply:
That's not correct. There are assertions that the methods are called a specific number of times.

Reply:
Ah, I see.